server/eclipse-project/src/main/java/us/freeandfair/corla/endpoint/CVRExportUpload.java

```diff
@@ -237,6 +255,30 @@ public String endpoint(final Request the_request, final Response the_response) {
   }
   
   /**
+   * Count the uploaded CVRs in storage.
+   * 
+   * @return the number of uploaded CVRs, or -1 if the count could not 
+   * be determined.
+   */
+  private OptionalLong count() {
+    OptionalLong result = OptionalLong.empty();
+    
+    try {
+      Persistence.beginTransaction();
+      final Session s = Persistence.currentSession();
+      final Query<Long> query = 
+          s.createQuery("select count(1) from CastVoteRecord where record_type = '" + 
+                        RecordType.UPLOADED + "'", Long.class);
+      result = OptionalLong.of(query.getSingleResult());
+      Persistence.commitTransaction();
+    } catch (final PersistenceException e) {
+      // ignore
```

Comment:
As far as the persistence exception... well, the only thing we _can_ do there is return no results. It, like any other PersistenceException, is an unrecoverable exception (per Hibernate's docs).